File: anvil/src/test/java/trikita/anvil/CurrentViewTest.java
package trikita.anvil;

import android.view.ViewGroup;

import org.junit.Test;

import static org.junit.Assert.*;
import static trikita.anvil.BaseDSL.*;

public class CurrentViewTest extends Utils {
    @Test
    public void testCurrentView() {
        assertNull(Anvil.currentView());
        Anvil.mount(container, new Anvil.Renderable() {
            public void view() {
                assertTrue(Anvil.currentView() instanceof ViewGroup);
                v(MockLayout.class, new Anvil.Renderable() {
                    public void view() {
                        assertTrue(Anvil.currentView() instanceof MockLayout);
                        v(MockView.class, new Anvil.Renderable() {
                            public void view() {
                                assertTrue(Anvil.currentView() instanceof MockView);
                                attr("text", "bar");
                                MockView view = Anvil.currentView(); // should cast automatically
                                assertEquals("bar", view.getText());
                            }
                        });
                        assertTrue(Anvil.currentView() instanceof MockLayout);
                    }
                });
                assertTrue(Anvil.currentView() instanceof ViewGroup);
            }
        });
        assertNull(Anvil.currentView());
    }
}

File: anvil/src/test/kotlin/trikita/anvil/CurrentViewTest.kt
package trikita.anvil

import android.view.View
import android.view.ViewGroup
import dev.inkremental.dsl.android.view.ViewScope
import dev.inkremental.dsl.android.widget.FrameLayoutScope
import kotlin.test.*


class CurrentViewTest : Utils() {
    @Test
    fun testCurrentView() {
        assertNull(Anvil.currentView())
        Anvil.mount(container) {
            assertTrue(Anvil.currentView<View>() is ViewGroup)
            v<MockLayout, FrameLayoutScope>(FrameLayoutScope) {
                assertTrue(Anvil.currentView<View>() is MockLayout)
                v<MockView, ViewScope>(ViewScope) {
                    assertTrue(Anvil.currentView<View>() is MockView)
                    attr("text", "bar")
                    val view: MockView = Anvil.currentView() // should cast automatically
                    assertEquals("bar", view.text)
                }
                assertTrue(Anvil.currentView<View>() is MockLayout)
            }
            assertTrue(Anvil.currentView<View>() is ViewGroup)
        }
        assertNull(Anvil.currentView())
    }
}
